src/bucket.js

```diff
@@ -2211,11 +2222,15 @@ Bucket.prototype.setUserProject = function(userProject) {
  * region_tag:storage_upload_encrypted_file
  * Example of uploading an encrypted file:
  */
-Bucket.prototype.upload = function(localPath, options, callback) {
+Bucket.prototype.upload = function(pathString, options, callback) {
   if (global.GCLOUD_SANDBOX_ENV) {
     return;
   }
 
+  var isURL =
+    !fs.existsSync(path.basename(pathString)) &&
```

Comment:
Unfortunately for the simplicity of the code, let's stay away from the sync method and use `fs.exists` instead. I'm open to not using `fx.exists` at all, and instead just testing the beginning of the string for the http/https protocol. What do you think?